src/app/qgisapp.cpp

```diff
@@ -7997,13 +7997,18 @@ void QgisApp::editPaste( QgsMapLayer *destinationLayer )
 
     // now create new feature using pasted feature as a template. This automatically handles default
     // values and field constraints
-    QgsFeature newFeature = QgsVectorLayerUtils::createFeature( pasteVectorLayer, geom, dstAttr, &context );
-    pasteVectorLayer->addFeature( newFeature );
-    newIds << newFeature.id();
-
+    newFeatures << QgsVectorLayerUtils::createFeature( pasteVectorLayer, geom, dstAttr, &context );
     ++featureIt;
   }
 
+  pasteVectorLayer->addFeatures( newFeatures );
+  QgsFeatureIds newIds;
+  Q_FOREACH ( const QgsFeature &f, newFeatures )
```

Comment:
Can you remove the `Q_FOREACH` and use a `for` instead? See https://github.com/qgis/QGIS-Enhancement-Proposals/issues/103